src/core/Utils.js

```diff
@@ -147,9 +149,73 @@ function runPromiseSequence (functions, ...args) {
 //   return (!f && 'not a function') || (s && s[1] || 'anonymous')
 // }
 
+function isPreviewSupported (fileTypeSpecific) {
+  // list of images that browsers can preview
+  if (/^(jpeg|gif|png|svg|svg\+xml|bmp)$/.test(fileTypeSpecific)) {
+    return true
+  }
+  return false
+}
+
+function getArrayBuffer (file) {
+  return new Promise(function (resolve, reject) {
+    var reader = new FileReader()
+    reader.addEventListener('load', function (e) {
+      // e.target.result is an ArrayBuffer
+      resolve(e.target.result)
+    })
+    reader.addEventListener('error', function (err) {
+      console.error('FileReader error' + err)
+      reject(err)
+    })
+    // file-type only needs the first 4100 bytes
+    const chunk = file.data.slice(0, 4100)
```

Comment:
It's a bit strange that the function called `getArrayBuffer` only returns 4100 bytes. maybe it could take a `Blob` object instead of an Uppy file object, and let the caller slice it?

```js
const chunk = file.data.slice(0, 4100)
getArrayBuffer(chunk).then(...)
```